### ISSUE TITLE
Incorrect deletion of unused identifiers

### ISSUE THREAD

**Issue by vatrif** [truebit/xUnique#38]
When the script deletes unused identifiers it breaks the JSON structure by deleting only a single line and not a whole JSON object.

`Following lines were deleted because of invalid format or no longer being used:`
`F6DD48671553FA9700A11AA0 /* Localizable.strings in Resources */ = {isa = PBXBuildFile; fileRef = F6DD485D1553FA9700A11AA0 /* Localizable.strings */; };`
`F6DD48671553FA9700A11AA0 /* Localizable.strings in Resources */,`
`F6DD485D1553FA9700A11AA0 /* Localizable.strings */ = {`

While leaving:
`			isa = PBXVariantGroup;`
			`children = (`
			`);`
			`name = Localizable.strings;`
			`path = "/Users/xp/Documents/iphone-app/Resources/Localization/EE";`
			`sourceTree = "<absolute>";`
		`};`

**Comment by truebit** [comment]
yes, it's a known issue. 

Sent from my mobile device.

**Comment by frranck** [comment]
Hi @truebit 
Is it a known issue that has been fixed ?

**Comment by truebit** [comment]
there is an open PR saying fixed this. You could give it a try.

It's a major change, I have no time doing tests.

Sent from my mobile device.

**Comment by frranck** [comment]
I tried both the master version and [the PR version](https://github.com/truebit/xUnique/issues/38) and both broke my project.pbxproj :(
Do you know if your project supports files with similar names but located in directories for different targets ?

**Comment by truebit** [comment]
yes，it should support. could you send me your project file？

Sent from my mobile device.

**Comment by simonayzman** [comment]
This is still an issue. Any movement on this?

**Comment by truebit** [comment]
@simonayzman Nope. This fix would change the core method to do the replacement, it's a major change. But in `master` branch, I added more logs to indicate that maybe texts needs to be deleted manually

**Comment by simonayzman** [comment]
@truebit Do you have a roadmap or suggestions for how you would fix it? In case someone decides to take a crack at it.

**Comment by miriyas** [comment]
Is there anyone knows replacement of this project?

**Comment by truebit** [comment]
@simonayzman You could try to make a mark if you delete a line ends with character like `(` or `{`, then delete all below lines after that before reach the corresponding  `)` and `}`

I will try to fix it in one or two days. I would need some problematic project file to test.

**Comment by simonayzman** [comment]
Not 100% sure about the xcode project file format, but I'm imagine that you'd have to ensure that the parentheses or braces would be balanced.

Try creating a react native project and uniquifying the iOS project?

**Comment by truebit** [comment]
@simonayzman the leading white spaces are the same. I am working on it. would push to a test branch, you could try it later.

**Comment by truebit** [comment]
I have push a new branch https://github.com/truebit/xUnique/tree/block_remove to address this issue.
If anyone got this problem, try this test branch by running the `xUnique.py` file directly to check if it works.
I do not know if I missed other project file scenarios.

**Comment by simonayzman** [comment]
Looks good to me! Works perfectly on my "bad" file.
